flutter_appauth_platform_interface/lib/src/errors.dart

```diff
@@ -0,0 +1,146 @@
+    required this.errorUri,
+    required this.domain,
+    required this.rootCauseDebugDescription,
+    required this.errorDebugDescription,
+    required this.userDidCancel,
+  });
+
+  /// The type of error.
+  ///
+  /// On iOS: one of the domain values from [here](https://github.com/openid/AppAuth-iOS/blob/c89ed571ae140f8eb1142735e6e23d7bb8c34cb2/Sources/AppAuthCore/OIDError.m#L31).
+  /// On Android: one of the type codes from [here](https://github.com/openid/AppAuth-Android/blob/c6137b7db306d9c097c0d5763f3fb944cd0122d2/library/java/net/openid/appauth/AuthorizationException.java).
+  ///
+  /// It's recommended to not use this unless needed. In most cases, errors
+  /// can be handled using the [error] property.
+  final String? type;
+
+  /// An error code from the platform's AppAuth SDK.
+  ///
+  /// On iOS/macOS: depending on the error type, it will be values from [here](https://github.com/openid/AppAuth-iOS/blob/c89ed571ae140f8eb1142735e6e23d7bb8c34cb2/Sources/AppAuthCore/OIDError.h).
+  /// On Android: one of the codes defined [here](https://github.com/openid/AppAuth-Android/blob/c6137b7db306d9c097c0d5763f3fb944cd0122d2/library/java/net/openid/appauth/AuthorizationException.java#L158).
+  ///
+  /// It's recommended to not use this unless needed. In most cases, errors
+  /// can be handled using the [error] property.
+  final String? code;
+
+  /// Error from the authorization server.
+  ///
+  /// For 400 errors from the authorization server, this is corresponds to the
+  /// `error` parameter as defined in the OAuth 2.0 framework [here](https://datatracker.ietf.org/doc/html/rfc6749#section-5.2).
+  /// Otherwise a short error describing what happened.
+  final String? error;
+
+  /// Short, human readable error description.
+  ///
+  /// This may come from the authhorization server that it correspond to the
+  /// `error_description` parameter as defined in the OAuth
+  /// 2.0 [here](https://datatracker.ietf.org/doc/html/rfc6749#section-5.2).
+  /// Otherwise, this is populated by the underlying platform's AppAuth SDK.
+  final String? errorDescription;
+
+  /// Error URI from the authorization server.
+  ///
+  /// Corresponds to the `error_uri` parameter defined in the OAuth 2.0
+  /// framework [here](https://datatracker.ietf.org/doc/html/rfc6749#section-5.2)
+  ///
+  /// Only populated on Android.
+  final String? errorUri;
+
+  /// Error domain from the iOS AppAuth SDK.
+  ///
+  /// Only populated on iOS.
+  final String? domain;
+
+  /// A debug description of the error from the platform's AppAuth SDK
+  final String? errorDebugDescription;
+
+  /// A debug description of the underlying cause of the error from the
+  /// platform's AppAuth SDK
+  final String? rootCauseDebugDescription;
+
+  /// Whether or not this error is caused by user cancellation
+  ///
+  /// Will only be `true` if the user cancelled the authorization flow by
+  /// closing the browser prematurely.
+  final bool userDidCancel;
```

Comment:
Wanted to double check something I noticed whilst updating docs etc. If I'm not mistaken, this is redundant as there's an exception specific for catching a cancellation. Can you confirm?

Reply:
Technically yes you're right, but I did this intentionally since I can imagine cases of users wanting to pass the `FlutterAppAuthPlatformErrorDetails` object into a method responsible for handling all the various cases, it seemed cleaner to encapsulate everything on one object.

I don't have particularly strong feelings about this one though.

Reply:
Thanks I've updated PR to remove this directly so I can merge this sooner. Will do a prerelease after though am looking to make it be part of a major release. The reason for this is I'm looking to bump the minimum Flutter SDK requirement to help make it easier to maintain. Would also allow for any potential breaking changes around error codes that have been discussed before